http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -24,22 +24,63 @@ import org.broadinstitute.dsde.workbench.model._
 import spray.json.{RootJsonFormat, RootJsonReader, _}
 import ca.mrvisser.sealerate
 
+import scala.util.matching.Regex
+
+sealed trait ContainerRegistry extends Product with Serializable {
+  def regex: Regex
+}
+object ContainerRegistry {
+  final case object GCR extends ContainerRegistry {
+    def regex: Regex = "(.*gcr.io)\\/(.*)\\/(.*)(\\:?.*)$".r
```

Comment:
See https://cloud.google.com/container-registry/docs/overview for name formats. It seems like this would work.

Reply:
Nit: you could use string literals to avoid double escaping

Reply:
can you give an example

Reply:
```
"""(.*gcr.io)\/(.*)\/(.*)(\:?.*)$""".r
```
==
```
"(.*gcr.io)\\/(.*)\\/(.*)(\\:?.*)$".r
```